### PR TITLE
${when}, support for RawValue on Inner and Else

### PR DESCRIPTION
**Pull request by 304NotModified**
fixes https://github.com/NLog/NLog/issues/3139

### CODE REVIEW

**Comment by 304NotModified** [comment]
@snakefoot I checked your comments in the other PR. Your right. I think I have proper implemented it now.

File: src/NLog/LayoutRenderers/Wrappers/WhenLayoutRendererWrapper.cs [hunk 1]
```diff
@@ -83,10 +84,40 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
...
+        {
+            if (ShouldRenderInner(logEvent))
+            {
+                if (Inner == null)
+                {
+                    value = null;
+                    return false;
+                }
+
+                return Inner.TryGetRawValue(logEvent, out value);
```
Comment (snakefoot) [inline src/NLog/LayoutRenderers/Wrappers/WhenLayoutRendererWrapper.cs]: Could this also work:
```c#
value = null;
if (ShouldRenderInner(logEvent))
   return Inner?.TryGetRawValue(logEvent, out value) ?? false;
else
   return Else?.TryGetRawValue(logEvent, out value) ?? false;
```

Comment (304NotModified) [reply src/NLog/LayoutRenderers/Wrappers/WhenLayoutRendererWrapper.cs]: I think it should work, but a bit hard to read. But I see the problem of the verbose/duplicate code. Will create a helper and let's see if it's then better.

Comment (304NotModified) [reply src/NLog/LayoutRenderers/Wrappers/WhenLayoutRendererWrapper.cs]: PS: we need a helper


image

Comment (304NotModified) [reply src/NLog/LayoutRenderers/Wrappers/WhenLayoutRendererWrapper.cs]: added helper. Let me know what you think

**Pull request merged**